Submission flow no consent keys Submission Screen (EXPOSUREAPP-4064)

AndroidMedaGalaxy opened the pull request:
### Description
Submission Flow No Consent Keys Submission Screen

### Steps to reproduce
The other UIs are being worked upon. once they're done you will be able to get to this screen after Ihr Testergibnis screen in No Consent Flow

### Screenshot
![SubmissionKeysNoConsent](https://user-images.githubusercontent.com/54317407/100811834-71abe800-3433-11eb-8fe8-23a943b5a55c.gif)


Note: 

1. Do not worry about the navigation from Submission keys to default thank you screen
The navigation will be changed while marrying the #1738 
2. CountryList is not being retrieved in the current environment. that's why you don't see any list being populated there.


Navigation Flow
Submission Keys -> close button on top -> home screen
Submission Keys - > press Einverstanden - > Google ENF API prompt will show up -> can observe progressbar below - > Thank you screen once completed

### TODO
Navigation to and from this screen. will be done in a seprate PR

AndroidMedaGalaxy committed:
Andere Warnen screen added

AndroidMedaGalaxy committed:
Merge branch 'release/1.9.x' into feature/4064-key-submission-no-consent

AndroidMedaGalaxy committed:
Merge conflict resolve and screen stability improvement

AndroidMedaGalaxy committed:
string text header added to ensure translation

janetback reviewed (dismissed):
UA approved

ralfgehrer reviewed (commented):
> CountryList is not being retrieved in the current environment. that's why you don't see any list being populated there.

Don't we have a default country in any case? E.g. Germany?

chiljamgossow commented on Corona-Warn-App/src/main/res/layout/include_submission_positive_other_warning_no_consent.xml:
```diff
@@ -0,0 +1,156 @@
+
+        <variable
+            name="viewModel"
+            type="de.rki.coronawarnapp.ui.submission.qrcode.consent.SubmissionConsentViewModel" />
```
Please use the fragments's own view model

chiljamgossow commented on Corona-Warn-App/src/main/res/values-de/legal_strings.xml:
```diff
@@ -34,4 +34,6 @@
     <!-- YTXT: Body for your consent screen agreement share symptoms -->
     <string name="submission_your_consent_agreement_share_symptoms">"<b>Wenn Sie zusätzlich Angaben zum Beginn Ihrer Symptome machen, werden auch diese geteilt.</b>\n\nSie können Ihr Einverständnis zurücknehmen, indem Sie oben „Andere warnen“ deaktivieren."</string>
+    <!-- YTXT: Body for keys submission no consent text -->
+    <string name="submission_no_consent_your_consent_subsection_body">"Durch Antippen von „Einverstanden“ willigen Sie wie folgt ein: \n\n<b>Die App teilt Ihr Testergebnis, um Nutzer von offiziellen Corona-Apps, denen Sie begegnet sind, zu warnen. Die Warnung funktioniert in den oben genannten Ländern. Wenn Sie zusätzlich Angaben zum Beginn Ihrer Symptome machen, werden auch diese geteilt.</b> Bi ...
```
There are too many line breaks, the space between paragraphs is too big.

ritsam replied:
what do you recon here then? seperate texts for each line?

AlexanderAlferov replied:
I think line breaks are okay. We are using them in many screens for now.

ritsam commented:
> > CountryList is not being retrieved in the current environment. that's why you don't see any list being populated there.
> 
> Don't we have a default country in any case? E.g. Germany?

If I keep the environment as INT the default country comes up. weird that it's not in WRU-XD. I'll investigate.

AndroidMedaGalaxy committed:
Merge branch 'release/1.9.x' into feature/4064-key-submission-no-consent

AndroidMedaGalaxy committed:
using submitTaskState instead of API request state

AndroidMedaGalaxy committed:
onDataPrivacyClick() function added in the relevant viewModel

AndroidMedaGalaxy committed:
long strings divided

AndroidMedaGalaxy committed:
legal strings updated

AndroidMedaGalaxy committed:
Merge branch 'release/1.9.x' into feature/4064-key-submission-no-consent

AndroidMedaGalaxy committed:
added navigation to mainfragment
removed unnecessary cases
fixed colors for under16 layout
fixed country list in layout
added onDataPrivacyClick method in own ViewModel
refactored color value

AndroidMedaGalaxy committed:
Merge branch 'feature/4064-key-submission-no-consent' of https://github.com/corona-warn-app/cwa-app-android into feature/4064-key-submission-no-consent

AndroidMedaGalaxy committed:
Merge branch 'release/1.9.x' into feature/4064-key-submission-no-consent

AndroidMedaGalaxy committed:
Merge branch 'release/1.9.x' into feature/4064-key-submission-no-consent

chiljamgossow commented on Corona-Warn-App/src/main/res/navigation/nav_graph.xml:
```diff
@@ -247,6 +247,25 @@
+        <action
+            android:id="@+id/action_submissionResultPositiveOtherWarningNoConsentFragment_to_submissionDoneFragment"
+            app:destination="@id/submissionDoneFragment"
+            app:popUpTo="@id/submissionDoneFragment"
```
You need to pop to homeFragment ( Inclusive = false) here to clear the back stack

ritsam replied:
Done

AndroidMedaGalaxy committed:
minor changes in navigation
